### PR TITLE
Added CVE-2020-10739

### PR DESCRIPTION
**Pull request by tausif-rh**

### CODE REVIEW

File: 2020/10xxx/CVE-2020-10739.json [hunk 1]
```diff
@@ -4,15 +4,74 @@
...
+                "refsource": "CONFIRM"
+            }
+        ]
     },
     "description": {
         "description_data": [
             {
                 "lang": "eng",
-                "value": "** RESERVED ** This candidate has been reserved by an organization or individual that will use it when announcing a new security problem. When the candidate has been publicized, the details for this candidate will be provided."
+                "value": "A null pointer exception flaw was found in servicemesh-proxy. When running Telemetry v2 (not on by default in version 1.4.x), an attacker could send a specially crafted packet to the ingress gateway or proxy sidecar, triggering a denial of service."
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:63]: Please mention the product `istio` in the description.  Is `servicemesh-proxy` a component of `istio`?

Please mention in the description affected versions and fixed versions (versions 1.4.x before 1.4.9 and versions 1.5.x before 1.5.4).

File: 2020/10xxx/CVE-2020-10739.json [hunk 2]
```diff
@@ -4,15 +4,74 @@
...
+                            {
+                                "product_name": "istio/envoy",
+                                "version": {
+                                    "version_data": [
+                                        {
+                                            "version_value": "1.5.3"
+                                        },
+                                        {
+                                            "version_value": "1.4.9"
+                                        }
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:25]: According the upstream security bulletin I think these should be updated to reflect that:

 - `>=1.4, <1.4.9`
 - `>=1.5, <1.5.4`

**Comment by cristina479** [comment]
Dear darunesh-RH,
Please address the comments made by iamleot. Thank you.

File: 2020/10xxx/CVE-2020-10739.json (hunk 2 above)
Comment (tausif-rh) [reply 2020/10xxx/CVE-2020-10739.json]: https://github.com/CVEProject/cvelist/pull/3826/commits/484d9a483ed252bdeb448dac0ded537e25c02c3a Fixed. Please review

File: 2020/10xxx/CVE-2020-10739.json [hunk 3]
```diff
@@ -46,21 +46,28 @@
...
+                "refsource": "MISC"
             }
         ]
     },
     "description": {
         "description_data": [
             {
                 "lang": "eng",
-                "value": "A null pointer exception flaw was found in servicemesh-proxy. When running Telemetry v2 (not on by default in version 1.4.x), an attacker could send a specially crafted packet to the ingress gateway or proxy sidecar, triggering a denial of service."
+                "value": "Istio 1.4.x to 1.4.8 with telemetry v2 enabled and Istio 1.5.x to 1.5.3 contain the following vulnerability when telemetry v2 is enabled. By sending a specially crafted packet, an attacker could trigger a Null Pointer Exception resulting in a Denial of Service. This could be sent to the ingress gateway or a sidecar, triggering a null pointer exception which results in a denial of service. This also affects servicemesh-proxy where a null pointer exception flaw was found in servicemesh-proxy. When running Telemetry v2 (not on by default in version 1.4.x), an attacker could send a specially crafted packet to the ingress gateway or proxy sidecar, triggering a denial of service."
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:70]: I would change the description from:

> "Istio 1.4.x to 1.4.8 with telemetry v2 enabled and Istio 1.5.x to 1.5.3 contain the following vulnerability when telemetry v2 is enabled. By sending a specially crafted packet, an attacker could trigger a Null Pointer Exception resulting in a Denial of Service. [...]

to:

> Istio 1.4.x before 1.4.9 and Istio 1.5.x before 1.5.4 contain the following vulnerability when telemetry v2 is enabled: By sending a specially crafted packet, an attacker could trigger a Null Pointer Exception resulting in a Denial of Service. [...]

This improve a bit the wording (also at the end it's written that Telemetry is not on by default in 1.4.x, no need to specify it also in the first sentence) and point out that fixed versions are released.

File: 2020/10xxx/CVE-2020-10739.json [hunk 4]
```diff
@@ -4,24 +4,24 @@
...
+                    "vendor_name": "istio.io",
                     "product": {
                         "product_data": [
                             {
                                 "product_name": "istio/envoy",
                                 "version": {
                                     "version_data": [
                                         {
-                                            "version_value": "1.5.3"
+                                            "version_value": "1.4.x to 1.4.8"
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:21]: Please mention *fixed* version, please change it to:

 `>= 1.4.0, < 1.4.9`

File: 2020/10xxx/CVE-2020-10739.json [hunk 5]
```diff
@@ -4,24 +4,24 @@
...
                                         },
                                         {
-                                            "version_value": "1.4.9"
+                                            "version_value": "1.5.x to 1.5.3"
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:24]: Please mention *fixed* version, please change it to:

 `>= 1.5.0, < 1.5.3`

File: 2020/10xxx/CVE-2020-10739.json (hunk 2 above)
Comment (iamleot) [reply 2020/10xxx/CVE-2020-10739.json]: @tausif-rh not all changes suggested were addressed.  I have reviewed and resuggested missed one and asked to improve a bit the wording in the description.

File: 2020/10xxx/CVE-2020-10739.json (hunk 5 above)
Comment (iamleot) [reply 2020/10xxx/CVE-2020-10739.json:24]: Please note that for 1-2 minutes I made a pasto and then modified it (it should be < 1.5.4, *not* < 1.5.3!, sorry!).

File: 2020/10xxx/CVE-2020-10739.json [hunk 6]
```diff
@@ -67,7 +67,7 @@
...
+                "value": "Istio 1.4.x before 1.4.9 and Istio 1.5.x before 1.5.4 contain the following vulnerability when telemetry v2 is enabled: By sending a specially crafted packet, an attacker could trigger a Null Pointer Exception resulting in a Denial of Service. This could be sent to the ingress gateway or a sidecar, triggering a null pointer exception which results in a denial of service. This also affects servicemesh-proxy where a null pointer exception flaw was found in servicemesh-proxy. When running Telemetry v2 (not on by default in version 1.4.x), an attacker could send a specially crafted packet to the ingress gateway or proxy sidecar, triggering a denial of service."
```
Comment (iamleot) [inline 2020/10xxx/CVE-2020-10739.json:70]: Only another minor nit, please change the `By' to `by' (the `.` was removed, no need to uppercase the `b` in `by`).

Comment (iamleot) [reply 2020/10xxx/CVE-2020-10739.json:70]: And, sorry, I was blameful for that: it was my copypasto.

**Pull request merged**